lesson4/hometask.rb

```diff
@@ -0,0 +1,23 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    fetch(method_name).call
+    self[method_name] = fetch(method_name).call
```

Comment:
Oh, almost forgot! There is one more thing which you have to handle: define custom `#respond_to_missing?` to modify result of `#respond_to?`. As you should remember from [lesson #4](https://github.com/nonkor/Betlab-ruby-courses/blob/master/lesson4/respond_to_missing.rb), every time when you redefine `#method_missing` you need to update `#respond_to_missing?` too.